lib/cn/csched_sp3.c

```diff
@@ -1584,7 +1584,7 @@ sp3_check_roots(struct sp3 *sp, uint qnum)
         if (sp3_work(spn, wtype_root, &sp->thresh, debug, &sp->wp))
             return false;
 
-        have_work = sp->wp && sp->wp->cw_action != CN_ACTION_NONE;
+        have_work = sp->wp->cw_action != CN_ACTION_NONE;
```

Comment:
The work struct ptr can be NULL in some cases can't it?  Or maybe that's old behavior.

Reply:
I neutered the only case where it could be NULL (maint-disabled).  In that case if maint is disabled sp3_work() now returns EAGAIN and we bail from the loop (same as for any error, then try again later).